indico/modules/events/surveys/controllers/display.py

```diff
@@ -84,6 +89,7 @@ def _checkParams(self, params):
                        .options(joinedload('submissions'),
                                 joinedload('sections').joinedload('children'))
                        .one())
+        self.submission = SurveySubmission.find_first(user=session.user, survey=self.survey)
```

Comment:
```python
if session.user:
    self.submission = SurveySubmission.query.with_parent(self.survey).filter_by(user=session.user).first()
else:
    self.submission = None
```

Otherwise you might end up getting an anonymous submission when you aren't logged in. Also, we kind of stopped using the `find_*` shortcuts in favor of plain sqlalchemy functionality some time ago.